java/ql/src/semmle/code/java/dataflow/internal/DataFlowImpl.qll

```diff
@@ -783,11 +836,33 @@ private predicate flowCandFwd(Node node, boolean fromArg, AccessPathFront apf, C
   if node instanceof CastingNode then compatibleTypes(node.getType(), apf.getType()) else any()
 }
 
+/** A node that should have a `nil` access path front. */
```

Comment:
This could be clarified to: "A node that requires an empty access path and should have its tracked type (re-)computed. This is either a source or a node reached through an additional step."